src/main/java/org/jabref/gui/fieldeditors/KeywordsEditor.java

```diff
@@ -1,24 +1,152 @@
+import org.jabref.gui.util.ViewModelListCellFactory;
 import org.jabref.logic.integrity.FieldCheckers;
+import org.jabref.logic.l10n.Localization;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.Keyword;
 import org.jabref.model.entry.field.Field;
 import org.jabref.preferences.PreferencesService;
 
-public class KeywordsEditor extends SimpleEditor implements FieldEditorFX {
+import com.airhacks.afterburner.views.ViewLoader;
+import com.dlsc.gemsfx.ChipView;
+import com.dlsc.gemsfx.TagsField;
+import jakarta.inject.Inject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class KeywordsEditor extends HBox implements FieldEditorFX {
+    private static final Logger LOGGER = LoggerFactory.getLogger(KeywordsEditor.class);
+
+    @FXML private TagsField<Keyword> keywordTagsField;
+
+    @Inject private PreferencesService preferencesService;
+    @Inject private DialogService dialogService;
+    @Inject private UndoManager undoManager;
+    @Inject private ClipBoardManager clipBoardManager;
+    @Inject private KeyBindingRepository keyBindingRepository;
+
+    private final KeywordsEditorViewModel viewModel;
 
     public KeywordsEditor(Field field,
                           SuggestionProvider<?> suggestionProvider,
-                          FieldCheckers fieldCheckers,
-                          PreferencesService preferences,
-                          UndoManager undoManager) {
-        super(field, suggestionProvider, fieldCheckers, preferences, undoManager);
+                          FieldCheckers fieldCheckers) {
+
+        ViewLoader.view(this)
+                  .root(this)
+                  .load();
+
+        this.viewModel = new KeywordsEditorViewModel(
+                field,
+                suggestionProvider,
+                fieldCheckers,
+                preferencesService,
+                undoManager);
+
+        keywordTagsField.setCellFactory(new ViewModelListCellFactory<Keyword>().withText(Keyword::toString));
+        keywordTagsField.setTagViewFactory(this::createTag);
+
+        keywordTagsField.setSuggestionProvider(request -> viewModel.getSuggestions(request.getUserText()));
+        keywordTagsField.setConverter(viewModel.getStringConverter());
+        keywordTagsField.setMatcher((keyword, searchText) -> keyword.get().toLowerCase().startsWith(searchText.toLowerCase()));
+        keywordTagsField.setComparator(Comparator.comparing(Keyword::get));
+
+        keywordTagsField.setNewItemProducer(searchText -> viewModel.getStringConverter().fromString(searchText));
+
+        keywordTagsField.setShowSearchIcon(false);
+        keywordTagsField.getEditor().getStyleClass().clear();
+        keywordTagsField.getEditor().getStyleClass().add("tags-field-editor");
+
+        Bindings.bindContentBidirectional(keywordTagsField.getTags(), viewModel.keywordListProperty());
+    }
+
+    private Node createTag(Keyword keyword) {
+        ChipView<String> chip = new ChipView<>();
```

Comment:
As we use `Labels` for the crossref field, I would suggest to use them here too to keep the UI consistent.

Reply:
Sure, I will undo this change and use `Labels` again.